File: Programmers/cpp/LEVEL_1_숫자_문자열과_영단어.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(string s) {
	int answer = 0;
	string strAnswer;
	unordered_map<string, char> um;

	string numbers[] = { "zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine" };

	for (int i = 0; i < 10; i++) {
		um[numbers[i]] = (char)(i + '0');
	}

	string temp;
	for (int i = 0; i < s.length(); i++) {
		if (s[i] >= '0' && s[i] <= '9') {
			if (!temp.empty()) {
				strAnswer += um[temp];
				temp.clear();
			}

			strAnswer += s[i];
			continue;
		}

		temp += s[i];

		if (um.count(temp)) {
			strAnswer += um[temp];
			temp.clear();
		}
	}

	if (!temp.empty()) {
		strAnswer += um[temp];
		temp.clear();
	}

	answer = stoi(strAnswer);

	return answer;
}

File: Programmers/java/LEVEL_1_숫자_문자열과_영단어.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int answer = 0;
        StringBuilder strAnswer = new StringBuilder();
        Map<String, Character> map = new HashMap<>();

        String[] numbers = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};

        for (int i = 0; i < 10; i++) {
            map.put(numbers[i], (char)(i + '0'));
        }

        StringBuilder temp = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) >= '0' && s.charAt(i) <= '9') {
                if (temp.length() != 0) {
                    strAnswer.append(map.get(temp.toString()));
                    temp.setLength(0);
                }

                strAnswer.append(s.charAt(i));
                continue;
            }

            temp.append(s.charAt(i));

            if (map.containsKey(temp.toString())) {
                strAnswer.append(map.get(temp.toString()));
                temp.setLength(0);
            }
        }

        if (temp.length() != 0) {
            strAnswer.append(map.get(temp.toString()));
            temp.setLength(0);
        }

        answer = Integer.parseInt(strAnswer.toString());

        return answer;
    }
}